src/key_gen/tests.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::dev_utils::{create_ids, new_common_rng, PeerId, RngChoice};
+use crate::id::SecretId;
+use crate::key_gen::{message::DkgMessage, DkgPhases, KeyGen};
+use maidsafe_utilities::{serialisation::serialise, SeededRng};
+use rand::Rng;
+use std::collections::{BTreeMap, BTreeSet};
+
+// Alter the seed here to reproduce failures.
+static SEED: RngChoice = RngChoice::SeededRandom;
```

Comment:
We really don't need to test rng's here. Let's not use seeded rngs everywhere. We don't need this and it's extra fluff. Let's just use rng direct and test this well. If the threshold_crypto crate wants to test the algo they can use a seeded rng, we should not need this

Reply:
hmm...  I think this is just for re-producing a failure so that will be easier for tracking down the flow and debug? This is not for test rng or threshold_crypto ?

Reply:
Can you give one case where the exact seed would help this? WE see rng and we immediately think to SEED it, but that IMO is a very false thing to do unless you don't trust the upstream lib and in that case we need to choose better libs or write tests for them. 
In this case seeding the RNG does not help debug at all. Happy to be proven wrong though ;-)

Reply:
It's the same for any crypto lib we use really. We don't need to test seeds there IMO

Reply:
We use rng to generate IDs. The same IDs among different test executions ease us to compare the logs and to track down certain behaviour pattern.

Reply:
I cannot see that being necessary or useful at all given the added complexity. The dkg algo shoud not care about specific id's at all so forcing that is at best misleading IMO

Reply:
Ok then, I will try to remove it.

Reply:
Thanks @maqi  Anything to make life simpler is my key right now, so we can move fast, very fast ;-)

Reply:
so far the rngs in tests shall not influence the test outcome.
the only concern I have is of a consistent log file (with same ID names) to track down.
However, if it does turn out to be a realy headache, we can add the SeededRng back later on ?